Content.Server/Xenoarchaeology/XenoArtifacts/Effects/Systems/PortalArtifactSystem.cs

```diff
@@ -21,22 +27,36 @@ public override void Initialize()
     private void OnActivate(Entity<PortalArtifactComponent> artifact, ref ArtifactActivatedEvent args)
     {
         var map = Transform(artifact).MapID;
-        var firstPortal = Spawn(artifact.Comp.PortalProto, _transform.GetMapCoordinates(artifact));
+        var validMinds = new List<EntityUid>();
+        var mindCont = new HashSet<Entity<MindContainerComponent>>();
 
-        var minds = new List<EntityUid>();
-        var mindQuery = EntityQueryEnumerator<MindContainerComponent, TransformComponent>();
-        while (mindQuery.MoveNext(out var uid, out var mc, out var xform))
+        _lookup.GetEntitiesOnMap<MindContainerComponent>(map, mindCont);
+        foreach (var mind in mindCont)
         {
-            if (mc.HasMind && xform.MapID == map)
-                minds.Add(uid);
+            // check if the MindContainer has a Mind and if the entity is not in a container (this also auto excludes AI)
+            if (mind.Comp.HasMind && !_container.IsEntityOrParentInContainer(mind))
+            {
+                validMinds.Add(mind.Owner);
+            }
         }
 
-        var target = _random.Pick(minds);
-        var secondPortal = Spawn(artifact.Comp.PortalProto, _transform.GetMapCoordinates(target));
+        //this would only be 0 if there were a station full of AIs and no one else
+        if (validMinds.Count != 0)
+        {
+            var firstPortal = Spawn(artifact.Comp.PortalProto, _transform.GetMapCoordinates(artifact));
+
+            var target = _random.Pick(validMinds);
+
+            var secondPortal = Spawn(artifact.Comp.PortalProto, _transform.GetMapCoordinates(target));
 
-        //Manual position swapping, because the portal that opens doesn't trigger a collision, and doesn't teleport targets the first time.
-        _transform.SwapPositions(target, secondPortal);
+            //Manual position swapping, because the portal that opens doesn't trigger a collision, and doesn't teleport targets the first time.
+            _transform.SwapPositions(target, secondPortal);
 
-        _link.TryLink(firstPortal, secondPortal, true);
+            _link.TryLink(firstPortal, secondPortal, true);
+        }
+        else
+        {
+            _adminLogger.Add(LogType.Teleport, $"Portal Artifact {ToPrettyString(artifact.Owner)} failed to find valid Teleport target");
```

Comment:
Admin logs are for information relevant to admins, not for debugging. Remove this.